crates/sui-core/src/traffic_controller/mod.rs

```diff
@@ -772,3 +820,15 @@ impl TrafficSim {
         );
     }
 }
+
+pub fn parse_ip(ip: &str) -> Option<IpAddr> {
+    ip.parse::<IpAddr>().ok().or_else(|| {
+        ip.parse::<SocketAddr>()
+            .ok()
+            .map(|socket_addr| socket_addr.ip())
+            .or_else(|| {
+                error!("Failed to parse value of {:?} to ip address or socket.", ip,);
```

Comment:
```suggestion
                error!("Failed to parse value of {:?} to ip address and socket.", ip);
```

Reply:
Actually this is correct. The parsing can take either an ip address or a socket object (ip + port). Will fix the trailing comma in the other PR since it's minor and want to get this landed